File: lang/java/ipc/src/main/java/org/apache/aingle/ipc/LocalTransceiver.java
/*

 */
package org.apache.aingle.ipc;

import java.io.IOException;
import java.nio.ByteBuffer;
import java.util.List;

/** Implementation of IPC that remains in process. */
public class LocalTransceiver extends Transceiver {
  private Responder responder;

  public LocalTransceiver(Responder responder) {
    this.responder = responder;
  }

  @Override
  public String getRemoteName() {
    return "local";
  }

  @Override
  public List<ByteBuffer> transceive(List<ByteBuffer> request) throws IOException {
    return responder.respond(request);
  }

  @Override
  public List<ByteBuffer> readBuffers() throws IOException {
    throw new UnsupportedOperationException();
  }

  @Override
  public void writeBuffers(List<ByteBuffer> buffers) throws IOException {
    throw new UnsupportedOperationException();
  }
}

File: lang/csharp/src/apache/ipc/LocalTransceiver.cs
/**

 */
using System;
using System.Collections.Generic;
using System.IO;

namespace AIngle.ipc
{
    public class LocalTransceiver : Transceiver
    {
        private readonly Responder responder;

        public LocalTransceiver(Responder responder)
        {
            if (responder == null) throw new ArgumentNullException("responder");

            this.responder = responder;
        }

        public override string RemoteName
        {
            get { return "local"; }
        }

        public override IList<MemoryStream> Transceive(IList<MemoryStream> request)
        {
            return responder.Respond(request);
        }

        public override IList<MemoryStream> ReadBuffers()
        {
            throw new NotSupportedException();
        }

        public override void WriteBuffers(IList<MemoryStream> getBytes)
        {
            throw new NotSupportedException();
        }
    }
}